pkg/cluster/sno.go

```diff
@@ -0,0 +1,33 @@
+package cluster
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type SNOCheck interface {
```

Comment:
This SNO check requires an entire informer sync from `configv1informers.InfrastructureInformer`. This is the "old" style of setting up controllers and I didn't want to mix in more legacy controller logic into the startup. If you think we should still reuse this I can of course rewrite but then I think we should base the entire logic on the typed informer.

Reply:
I could also change it to check the Infrastructure CR like the informer function to read out the `infrastructureTopology` instead of starting the controller

Reply:
The function cannot be called once and gced because the informer needs to be started and synced (basically a controller with cache). There is no way to use the function without the informer. 

I can however prepare a PR to use the Check without the Informer but that will be hard to do because of way that its built into the client. We will need a check that is not relying on the inbuilt typed client but on controller-runtime.